frontend/src/components/MeetingRoom/ParticipantListItemMenu/ParticipantListItemMenu.spec.tsx

```diff
@@ -0,0 +1,88 @@
+import { beforeEach, describe, expect, it, Mock, vi } from 'vitest';
+import { act, render, screen } from '@testing-library/react';
+import { Subscriber } from '@vonage/client-sdk-video';
+import { SubscriberWrapper } from '../../../types/session';
+import ParticipantListItemMenu from '.';
+import useSessionContext from '../../../hooks/useSessionContext';
+import { SessionContextType } from '../../../Context/SessionProvider/session';
+
+vi.mock('../../../hooks/useSessionContext');
+
+const mockUseSessionContext = useSessionContext as Mock<[], SessionContextType>;
+
+describe('ParticipantListItem', () => {
+  let mockSessionContext: SessionContextType;
+
+  const mockSubscriberWrapper: SubscriberWrapper = {
+    id: 'subId',
+    isPinned: false,
+    isScreenshare: false,
+    subscriber: {} as Subscriber,
+    element: {} as HTMLVideoElement,
+  };
+  const defaultProps = {
+    participantName: 'John Doe',
+    subscriberWrapper: mockSubscriberWrapper,
+  };
+  beforeEach(() => {
+    mockSessionContext = {
+      isMaxPinned: false,
+      pinSubscriber: vi.fn(),
+    } as unknown as SessionContextType;
+    mockUseSessionContext.mockImplementation(() => mockSessionContext);
+  });
+
+  it('can pin participant', async () => {
+    render(<ParticipantListItemMenu {...defaultProps} />);
+    const button = screen.getByRole('button');
+    await act(() => button.click());
+    const pinButton = await screen.getByText('Pin John Doe');
+    await act(() => pinButton.click());
+    expect(mockSessionContext.pinSubscriber).toHaveBeenCalledWith('subId');
+  });
+
+  it('can unpin participant', async () => {
+    render(
+      <ParticipantListItemMenu
+        {...{ ...defaultProps, subscriberWrapper: { ...mockSubscriberWrapper, isPinned: true } }}
+      />
+    );
+    const button = screen.getByRole('button');
+    await act(() => button.click());
+    const pinButton = await screen.getByText('Unpin John Doe');
+    await act(() => pinButton.click());
+    expect(mockSessionContext.pinSubscriber).toHaveBeenCalledWith('subId');
+  });
+
+  it('cannot pin participant if isMaxPinned is true', async () => {
```

Comment:
nit: could we rename the name of the test from `if isMaxPinned is true` to something more self-explanatory such as `if maximum number of pinned tiles is reached`